build-support/bin/ci.sh

```diff
@@ -298,7 +278,14 @@ if [[ "${run_integration:-false}" == "true" ]]; then
     ./pants.pex --tag='+integration' test.pytest \
       "--test-pytest-test-shard=${python_intg_shard}" \
       src/python:: tests/python:: -- "${PYTEST_PASSTHRU_ARGS[@]}"
-  ) || die "Pants Integration test failure"
+  ) || die "Pants integration test failure"
+  (
+    # NB: We use test_testprojects_integration.py to specifically run the ones that
+    # we care about, and skip the rest.
+    ./pants.pex --tag='+integration' --exclude-target-regexp='.*/testprojects/.*' \
+      test.pytest "--test-pytest-test-shard=${python_intg_shard}" \
```

Comment:
Quoting like you do here (`"--test-pytest-test-shard=${python_intg_shard}"`) reads strangely. I'd expect `--test-pytest-test-shard="${python_intg_shard}"` or even `--test-pytest-test-shard=${python_intg_shard}` as a soft assertion that `${python_intg_shard}` contains no whitespace - which it can't by definition of the `test.pytest` option. This is not worth a whole new CI burn though! If there are other changes, consider this cleanup though.

Reply:
I have to pull master due to the zipkin PR getting merged half-way through CI causing a unit test to fail. Will improve!